src/Std/Time/Duration.lean

```diff
@@ -0,0 +1,132 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Sofia Rodrigues
+-/
+prelude
+import Std.Time.DateTime
+import Std.Time.Zoned
+
+namespace Std
+namespace Time
+open Internal
+
+set_option linter.all true
+
+/--
+Duration is just a period between two timestamps.
+-/
+def Duration := Timestamp
+  deriving Repr, BEq
+
+instance : ToString Duration where
+  toString s :=
+    let (sign, secs, nanos) :=
+      if s.second.val > 0 then ("" ,s.second.val, s.nano.val)
+      else if s.second.val < 0 then ("-", -s.second.val, -s.nano.val)
+      else if s.nano.val < 0 then ("-", -s.second.val, -s.nano.val) else ("", s.second.val, s.nano.val)
+    sign ++ toString secs ++ "." ++ toString nanos ++ "s"
+
+instance : Repr Duration where
+  reprPrec s := reprPrec (toString s)
+
+namespace Duration
+
+/--
+Converts a `Duration` to a `Minute.Offset`
+-/
+def toMinutes (tm : Duration) : Minute.Offset :=
+  tm.second.ediv 60
+
+/--
+Converts a `Duration` to a `Day.Offset`
+-/
+def toDays (tm : Duration) : Day.Offset :=
+  tm.second.ediv 86400
+
+/--
+Creates a new `Duration` out of `Second.Offset`.
+-/
+def ofSeconds (secs : Second.Offset) : Duration :=
+  Timestamp.ofSeconds secs
+
+/--
+Creates a new `Duration` out of `Nanosecond.Offset`.
+-/
+def ofNanosecond (secs : Nanosecond.Offset) : Duration :=
+  Timestamp.ofNanoseconds secs
+
+/--
+Calculates a `Duration` out of two `Timestamp`s.
+-/
+def since (f : Timestamp) : IO Duration := do
+  let cur ← Timestamp.now
+  return Std.Time.Timestamp.sub cur f
+
+/--
+Adds a `Duration` to a `Timestamp`.
+-/
+def add (f : Timestamp) (d : Duration) : Timestamp :=
+  Std.Time.Timestamp.add f d
+
+/--
+Checks if the duration is zero seconds ands zero nanoseconds.
+-/
+def isZero (d : Duration) : Bool :=
+  d.second.val = 0 ∧ d.nano.val = 0
+
+instance : HAdd Timestamp Duration Timestamp where
+  hAdd := add
+
+instance : HSub Timestamp Timestamp Duration where
+  hSub := Std.Time.Timestamp.sub
+
+end Duration
+
+namespace PlainDate
+
+/--
+Calculates the duration between a given `PlainDate` and a specified date.
+-/
+def since [ToTimestamp α] (date : PlainDate) (since : α) : Duration :=
```

Comment:
These functions seem like it is very easy to accidentally get the argument order wrong; there should be an example in the docstring at least.

Reply:
Added examples.